keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,2346 @@
+	// considering the needs of Pods that will use the ResourceClaim
+	// because those Pods are not known yet.
+	AllocationModeImmediate AllocationMode = "Immediate"
+)
+
+// ResourceClaimStatus tracks whether the resource has been allocated and what
+// the resulting attributes are.
+type ResourceClaimStatus struct {
+	// DriverName is a copy of the driver name from the ResourceClass at
+	// the time when allocation started. It's necessary to support
+	// deallocation when the class gets deleted before a claim.
+	DriverName string
+
+	// Scheduling contains information that is only relevant while the
+	// scheduler and the resource driver are in the process of selecting a
+	// node for a Pod and the allocation mode is AllocationModeDelayed. The
+	// resource driver should unset this when it has successfully allocated
+	// the resource.
+	Scheduling SchedulingStatus
+
+	// Allocation is set by the resource driver once a resource has been
+	// allocated successfully. Nil indicates that the resource is not
+	// allocated.
+	Allocation *AllocationResult
+
+	// Deallocate may be set to true to request deallocation of a resource as soon
+	// as it is unused. The scheduler uses this when it finds that deallocating
+	// the resource and reallocating it elsewhere might unblock a pod.
+	//
+	// The resource driver checks this fields and resets it to false
+	// together with clearing the Allocation field. It also sets it
+	// to false when the resource is not allocated.
+	Deallocate bool
+
+	// ReservedFor indicates which entities are currently allowed to use
+	// the resource.  Usually those are Pods, but other objects are
+	// also possible as long as they exist. The resource controller will
+	// remove all entries for objects that do not exist.
+	//
+	// A scheduler must add a Pod that it is scheduling. This must be done
+	// in an atomic ResourceClaim update because there might be multiple
+	// schedulers working on different Pods that compete for access to the
+	// same ResourceClaim.
+	//
+	// kubelet will check this before allowing a Pod to run because a
+	// a user might have selected a node manually without reserving
+	// resources or a
+	// scheduler might have missed that step, for example because it
+	// doesn't support dynamic resource allocation or the feature was
+	// disabled.
+	ReservedFor []metav1.OwnerReference
+}
+
+// SchedulingStatus contains information that is relevant while
+// a Pod with delayed allocation is being scheduled.
+type SchedulingStatus struct {
+	// When allocation is delayed, the scheduler must set
+	// the node for which it wants the resource to be allocated
+	// before the driver proceeds with allocation.
+	//
+	// This field may only be set by a scheduler, but not get
+	// overwritten. It will get reset by the driver when allocation
+	// succeeds or fails. This ensures that different schedulers
+	// that handle different Pods do not accidentally trigger
```

Comment:
How?

Not that this needs to be answered in the API docs, but it can go in the design details.

Reply:
With `SelectedNode` in ResourceClaimStatus, only one scheduler succeeds with setting it. The other one then isn't allowed to overwrite with a different value.

This will change when moving the field in PodScheduling.